pkg/secrets/clients/k8s/k8s_secrets_client.go

```diff
@@ -1,48 +1,50 @@
 package k8s
 
 import (
-	"bytes"
-
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 
 	"github.com/cyberark/secrets-provider-for-k8s/pkg/log"
 	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
-	"github.com/cyberark/secrets-provider-for-k8s/pkg/utils"
 )
 
-type RetrieveK8sSecretFunc func(namespace string, secretName string) (map[string][]byte, error)
-type PatchK8sSecretFunc func(namespace string, secretName string, stringDataEntriesMap map[string][]byte) error
+type RetrieveK8sSecretFunc func(namespace string, secretName string) (*v1.Secret, error)
+type UpdateK8sSecretFunc func(namespace string, secretName string, originalK8sSecrets *v1.Secret, stringDataEntriesMap map[string][]byte) error
 
-func RetrieveK8sSecret(namespace string, secretName string) (map[string][]byte, error) {
+func RetrieveK8sSecret(namespace string, secretName string) (*v1.Secret, error) {
 	// get K8s client object
 	kubeClient, _ := configK8sClient()
 	log.Info(messages.CSPFK005I, secretName, namespace)
 	k8sSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
 	if err != nil {
-		return nil, err
+		// Error messages returned from K8s should be printed only in debug mode
+		log.Debug(messages.CSPFK004D, err.Error())
+		return nil, log.RecordedError(messages.CSPFK020E)
 	}
 
-	return k8sSecret.Data, nil
+	return k8sSecret, nil
 }
 
-func PatchK8sSecret(namespace string, secretName string, stringDataEntriesMap map[string][]byte) error {
+func UpdateK8sSecret(namespace string, secretName string, originalK8sSecrets *v1.Secret, stringDataEntriesMap map[string][]byte) error {
 	// get K8s client object
 	kubeClient, _ := configK8sClient()
 
-	stringDataEntry, err := generateStringDataEntry(stringDataEntriesMap)
-	if err != nil {
-		return log.RecordedError(messages.CSPFK024E)
+	log.Info(messages.CSPFK006I, secretName, namespace)
```

Comment:
can you move this further down to be just before the actual action?

Reply:
done